src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -34,33 +50,61 @@
+    private final YugabyteDBConnectorConfig ybConnectorConfig;
     private final YugabyteDBSchema schema;
     private final SnapshotProgressListener snapshotProgressListener;
     private final YugabyteDBTaskContext taskContext;
     private final EventDispatcher<YBPartition,TableId> dispatcher;
     protected final Clock clock;
     private final Snapshotter snapshotter;
+    private final YugabyteDBConnection connection;
+
+    private final AsyncYBClient asyncClient;
+    private final YBClient syncClient;
+
+    private OpId lastCompletelyProcessedLsn;
+
+    private YugabyteDBTypeRegistry yugabyteDbTypeRegistry;
+
+    private final Metronome retryMetronome;
 
     public YugabyteDBSnapshotChangeEventSource(YugabyteDBConnectorConfig connectorConfig,
                                                YugabyteDBTaskContext taskContext,
-                                               Snapshotter snapshotter,
+                                               Snapshotter snapshotter, YugabyteDBConnection connection,
                                                YugabyteDBSchema schema, YugabyteDBEventDispatcher<TableId> dispatcher, Clock clock,
                                                SnapshotProgressListener snapshotProgressListener) {
         super(connectorConfig, snapshotProgressListener);
-        this.connectorConfig = connectorConfig;
+        this.ybConnectorConfig = connectorConfig;
         this.schema = schema;
         this.taskContext = taskContext;
         this.dispatcher = dispatcher;
         this.clock = clock;
         this.snapshotter = snapshotter;
+        this.connection = connection;
         this.snapshotProgressListener = snapshotProgressListener;
+
+        this.asyncClient = new AsyncYBClient.AsyncYBClientBuilder(connectorConfig.masterAddresses())
+            .defaultAdminOperationTimeoutMs(connectorConfig.adminOperationTimeoutMs())
+            .defaultOperationTimeoutMs(connectorConfig.operationTimeoutMs())
+            .defaultSocketReadTimeoutMs(connectorConfig.socketReadTimeoutMs())
+            .numTablets(connectorConfig.maxNumTablets())
+            .sslCertFile(connectorConfig.sslRootCert())
+            .sslClientCertFiles(connectorConfig.sslClientCert(), connectorConfig.sslClientKey())
+            .build();
+        
+        this.syncClient = new YBClient(this.asyncClient);
+
+        this.retryMetronome = Metronome.parker(
+            Duration.ofMillis(connectorConfig.connectorRetryDelayMs()), Clock.SYSTEM);
+
+        this.yugabyteDbTypeRegistry = taskContext.schema().getTypeRegistry();
     }
 
     @Override
     public SnapshotResult<YugabyteDBOffsetContext> execute(ChangeEventSourceContext context, YBPartition partition, YugabyteDBOffsetContext previousOffset)
             throws InterruptedException {
         SnapshottingTask snapshottingTask = getSnapshottingTask(partition, previousOffset);
+        LOGGER.info("Dispatcher in snapshot: " + dispatcher.toString());
```

Comment:
Better as DEBUG? I am assuming this can happen often

Reply:
Done.